index.js

```diff
@@ -14,18 +17,64 @@ module.exports = {
   },
 
   setupPreprocessorRegistry: function(type, registry) {
-    if (type !== 'self') return;
+    this.setupJsPreprocessing(registry, type);
+
+    if (type === 'self') {
```

Comment:
There are 2 registries: `self` is the registry for the addon itself, `parent` is for the app that is consuming the addon. For the css and hbs preprocessing, we only need it for the addon. For the js preprocessing, we need it for both since we need to generate the addon component _and_ the app component (which simply re-exports the addon to make it available to the app)